Just hit goroutines again, sadly:

```
1m 15s
1m 36s
Run go test -tags=llvm15 -short -skip=TestErrors
no Go files in /home/runner/work/tinygo/tinygo/tests/testing
--- FAIL: TestBuild (0.00s)
    --- FAIL: TestBuild/Host (0.00s)
        --- FAIL: TestBuild/Host/goroutines.go (2.90s)
            main_test.go:377: output did not match (expected 523 bytes, got 523 bytes):
            main_test.go:377: diff expected actual
                --- expected
                +++ actual
                @@ -10,8 +10,8 @@
                   wait end
                 end waiting
                 value produced after some time: 42
                -non-blocking goroutine
                 done with non-blocking goroutine
                +non-blocking goroutine
                 async interface method call
                 slept inside func pointer 8
                 slept inside closure, with value: 20 8
                
```

--- a/main_test.go
+++ b/main_test.go
@@ -280,6 +280,11 @@ func runPlatTests(options compileopts.Options, tests []string, t *testing.T) {
 	}
 
 	for _, name := range tests {
+		if name == "goroutines.go" && (spec.Scheduler == "threads" || spec.Scheduler == "cores") {
+			// This test intentionally checks concurrent scheduling by comparing
+			// output order, so only run it with non-threaded schedulers.
+			continue
+		}
 		if options.GOOS == "linux" && (options.GOARCH == "arm" || options.GOARCH == "386") {
 			switch name {
 			case "timers.go":

